File: application/models/abstract/AbstractCharacter.php
<?php
/**
 * Абстрактный класс персонажей
 * @author ismd
 */

abstract class AbstractCharacter extends PsAbstractModel {

    protected $_id;     // id в таблице Character (сам персонаж)
    protected $_idUser; // id в таблице User (пользователь)
    protected $_name;
    protected $_level;
    protected $_money;
    protected $_hp;
    protected $_maxHp;
    protected $_minDamage;
    protected $_maxDamage;
    protected $_experience;
    protected $_image;
    protected $_strength;
    protected $_dexterity;
    protected $_endurance;
    protected $_idMap;
    protected $_x;
    protected $_y;

    public function setId($value) {
        $this->_id = (int)$value;
        return $this;
    }

    public function getId() {
        return $this->_id;
    }

    public function setIdUser($value) {
        $this->_idUser = (int)$value;
        return $this;
    }

    public function getIdUser() {
        return $this->_idUser;
    }

    public function setName($value) {
        $this->_name = (string)$value;
        return $this;
    }

    public function getName() {
        return $this->_name;
    }

    public function setLevel($value) {
        $this->_level = (int)$value;
        return $this;
    }

    public function getLevel() {
        return $this->_level;
    }

    public function setMoney($value) {
        $this->_money = (int)$value;
        return $this;
    }

    public function getMoney() {
        return $this->_money;
    }

    public function setHp($value) {
        $this->_hp = (int)$value;
        return $this;
    }

    public function getHp() {
        return $this->_hp;
    }

    public function setMaxHp($value) {
        $this->_maxHp = (int)$value;
        return $this;
    }

    public function getMaxHp() {
        return $this->_maxHp;
    }

    public function setMinDamage($value) {
        $this->_minDamage = (int)$value;
        return $this;
    }

    public function getMinDamage() {
        return $this->_minDamage;
    }

    public function setMaxDamage($value) {
        $this->_maxDamage = (int)$value;
        return $this;
    }

    public function getMaxDamage() {
        return $this->_maxDamage;
    }

    public function setExperience($value) {
        $this->_experience = (int)$value;
        return $this;
    }

    public function getExperience() {
        return $this->_experience;
    }

    public function setImage($value) {
        $this->_image = (string)$value;
        return $this;
    }

    public function getImage() {
        return $this->_image;
    }

    public function setStrength($value) {
        $this->_strength = (int)$value;
        return $this;
    }

    public function getStrength() {
        return $this->_strength;
    }

    public function setDexterity($value) {
        $this->_dexterity = (int)$value;
        return $this;
    }

    public function getDexterity() {
        return $this->_dexterity;
    }

    public function setEndurance($value) {
        $this->_endurance = (int)$value;
        return $this;
    }

    public function getEndurance() {
        return $this->_endurance;
    }

    public function setIdMap($value) {
        $this->_idMap = (int)$value;
        return $this;
    }

    public function getIdMap() {
        return $this->_idMap;
    }

    public function setX($value) {
        $this->_x = (int)$value;
        return $this;
    }

    public function getX() {
        return $this->_x;
    }

    public function setY($value) {
        $this->_y = (int)$value;
        return $this;
    }

    public function getY() {
        return $this->_y;
    }
}

File: bin/server.common/src/game/models/common/AbstractCharacter.java
package game.models.common;

/**
 * @author ismd
 */
abstract public class AbstractCharacter {

    private int _id;
    private int _idUser;
    private String _name;
    private int _level;
    private int _money;
    private int _hp;
    private int _maxHp;
    private int _minDamage;
    private int _maxDamage;
    private int _experience;
    private String _image;
    private int _strength;
    private int _dexterity;
    private int _endurance;
    private int _idMap;
    private int _x;
    private int _y;

    public int getId() {
        return _id;
    }

    public AbstractCharacter setId(int id) {
        _id = id;
        return this;
    }

    public int getIdUser() {
        return _idUser;
    }

    public AbstractCharacter setIdUser(int idUser) {
        _idUser = idUser;
        return this;
    }

    public String getName() {
        return _name;
    }

    public AbstractCharacter setName(String name) {
        _name = name;
        return this;
    }

    public int getLevel() {
        return _level;
    }

    public AbstractCharacter setLevel(int level) {
        _level = level;
        return this;
    }

    public int getMoney() {
        return _money;
    }

    public AbstractCharacter setMoney(int money) {
        _money = money;
        return this;
    }

    public int getHp() {
        return _hp;
    }

    public AbstractCharacter setHp(int hp) {
        _hp = hp;
        return this;
    }

    public int getMaxHp() {
        return _maxHp;
    }

    public AbstractCharacter setMaxHp(int maxHp) {
        _maxHp = maxHp;
        return this;
    }

    public int getMinDamage() {
        return _minDamage;
    }

    public AbstractCharacter setMinDamage(int minDamage) {
        _minDamage = minDamage;
        return this;
    }

    public int getMaxDamage() {
        return _maxDamage;
    }

    public AbstractCharacter setMaxDamage(int maxDamage) {
        _maxDamage = maxDamage;
        return this;
    }

    public int getExperience() {
        return _experience;
    }

    public AbstractCharacter setExperience(int experience) {
        _experience = experience;
        return this;
    }

    public String getImage() {
        return _image;
    }

    public AbstractCharacter setImage(String image) {
        _image = image;
        return this;
    }

    public int getStrength() {
        return _strength;
    }

    public AbstractCharacter setStrength(int strength) {
        _strength = strength;
        return this;
    }

    public int getDexterity() {
        return _dexterity;
    }

    public AbstractCharacter setDexterity(int dexterity) {
        _dexterity = dexterity;
        return this;
    }

    public int getEndurance() {
        return _endurance;
    }

    public AbstractCharacter setEndurance(int endurance) {
        _endurance = endurance;
        return this;
    }

    public int getIdMap() {
        return _idMap;
    }

    public AbstractCharacter setIdMap(int idMap) {
        _idMap = idMap;
        return this;
    }

    public int getX() {
        return _x;
    }

    public AbstractCharacter setX(int x) {
        _x = x;
        return this;
    }

    public int getY() {
        return _y;
    }

    public AbstractCharacter setY(int y) {
        _y = y;
        return this;
    }
}
